src/https-server.ts

```diff
@@ -0,0 +1,186 @@
+import https from 'https'
+
+import { HttpIncomingMessage, HttpRequestListener } from './http-common'
+import { HttpServerBase } from './http-server-base'
+
+const localhostCertificate = `-----BEGIN CERTIFICATE-----
```

Comment:
I think these certificates should be moved to a separate file. Perhaps just a JS file exporting each certificate as a variable.